### DISCUSSION TITLE
Failed to export onnx model when I use kornia.geometry.transform.imgwarp.warp_perspective in the model forward function

### DISCUSSION THREAD

**Discussion by knavezl** [kornia/kornia#2992] [General]
my export model code is:
input_img = torch.randn(1 ,4, 3, 864, 1536).cuda() 
num_cameras = 4
num_classes = 3
resolution = [360 ,4 ,360]
Y, Z, X = resolution
encoder_name = 'res50'
model_params = torch.load(pt_path) 
state_dict = {}
for key in model_params["state_dict"].keys():
    state_dict_key = key.replace('model.' , '') 
    state_dict[state_dict_key] = model_params["state_dict"][key] 
model = MVDet(Y, Z, X, encoder_type = encoder_name, num_cameras = num_cameras, num_classes = num_classes)
model.load_state_dict(state_dict,strict=True)
model.cuda()  
torch.onnx.export(model, input_img  , onnx_path, verbose=False,opset_version=13)

The error is：
File "/home/user/BEV/TrackTacular/WorldTrack/models/mvdet.py", line 172, in forward
feat_mems_ = warp_perspective(feat_cams_, proj_mats, (self.Y, self.X), align_corners=False)
File "/home/user/anaconda3/envs/pytorch2.1/lib/python3.9/site-packages/kornia/geometry/transform/imgwarp.py", line 126, in warp_perspective
grid = transform_points(src_norm_trans_dst_norm[:, None, None], grid)
File "/home/user/anaconda3/envs/pytorch2.1/lib/python3.9/site-packages/kornia/geometry/linalg.py", line 191, in transform_points
trans_01 = torch.repeat_interleave(trans_01, repeats=points_1.shape[0] // trans_01.shape[0], dim=0)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper_CUDA__index_select)

Have you tried to transform the model and encountered the same problem? Do you have any solutions?

**Comment by edgarriba** [comment]
I personally haven't tried but would be great to support this. Ideally this test should be extend and validate: https://github.com/kornia/kornia/blob/main/tests/geometry/transform/test_imgwarp.py#L314

The function that crashes seems to be https://github.com/kornia/kornia/blob/main/kornia/geometry/linalg.py#L188

I can try to allocate some time for this but i have other priorities. Are you open to dig into the issue and propose a PR ?

**Reply by edgarriba** [reply]
ok, I think I see the issue -- i believe somewhere here we need to infer the src.device https://github.com/kornia/kornia/blob/main/kornia/geometry/transform/imgwarp.py#L114-L116

have to tried to run this with eager mode?

**Reply by knavezl** [reply]
In the code line : 
https://github.com/kornia/kornia/blob/main/kornia/geometry/linalg.py#L188  ,
I have tried to set repeats as a default num value ,the error "RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper_CUDA__index_select)" disappears ,
but another error has occurred：
torch.onnx.errors.SymbolicValueError: Unsupported: ONNX export of instance_norm for unknown channel size.

**Reply by edgarriba** [reply]
the last error i don't think it's anything related to this function. Looks more something coming from a model

**Reply by knavezl** [reply]
Thank you for your reply. I will try other methods again for the last error
